### ISSUE TITLE
sample_ids being ignored?

### ISSUE THREAD

**Issue by emanjavacas** [rdspring1/PyTorch_LM#4]
Hi! thanks for your code. I've been reading through it to understand the approach and I've noticed that the output of `sampled` is actually always a zero long-tensor:

https://github.com/rdspring1/PyTorch_GBW_LM/blob/master/lm/model.py#L68-L69

Is this the way is supposed to work? I was understanding that the sampled softmax obtains the speed up by computing the loss on only a sample of the entire vocabulary. But the way it's setup the loss would always be computed with respect to the same target (0).

Or is there something else I might be missing?

greetings!

**Comment by rdspring1** [comment]
Look at the previous line. The true labels and noise samples are concatenated together. Therefore, since the true label for all the examples is in the first column, they all have the same target value.

**Comment by emanjavacas** [comment]
Oh! I see it now. I had missed that the loss is computed based on which one is the true target assuming a set of the true target and negative examples. Thanks!
